# The Microsoft C Runtime source code to which this document refers is available
# directly from Microsoft Corporation, under a separate license.
# Please ensure that if you are using that source code, you have appropriate
# rights to use it.  By providing you access to this file, Mozilla Corporation
# and its affiliates do not purport to grant any rights in that source code. 
# Binaries are available under separate licenses at 
# http://www.microsoft.com/downloads/details.aspx?familyid=200b2fd9-ae1a-4a14-984d-389c36f85647&displaylang=en
# convert for VC9 SP1 by Takeshi Ichimaru
# Microsoft Visual C++ 2008 Redistributable Package (x86) Binaries are available under
# separate licenses at 
# http://www.microsoft.com/downloads/details.aspx?familyid=9B2DA534-3E03-4391-8A4D-074B9F2BC1BF&displaylang=en
diff -re crt/src/crt0.c crt-sp1/src/crt0.c
212c
        /*
         * this used to happen in _mtinit, but we need it before malloc
         */
        _init_pointers();       /* initialize global function pointers */

        if ( malloc_init_hard() )                /* initialize heap */
.
87a
extern BOOL malloc_init_hard(void);
.
diff -re crt/src/crt0dat.c crt-sp1/src/crt0dat.c
837d
826d
diff -re crt/src/crtdll.c crt-sp1/src/crtdll.c
31,40d
diff -re crt/src/crtexe.c crt-sp1/src/crtexe.c
321,323d
35,45d
diff -re crt/src/crtheap.c crt-sp1/src/crtheap.c
61c
    pv = calloc(count, size);
.
58,59d
diff -re crt/src/crtlib.c crt-sp1/src/crtlib.c
684,691d
355d
339a
                 malloc_shutdown();

.
298d
279d
249,250d
239d
226c
            /*
             * this used to happen in _mtinit, but we need it before malloc
             */
            _init_pointers();       /* initialize global function pointers */

            if ( malloc_init_hard() )   /* initialize heap */
.
43a
extern BOOL malloc_init_hard(void);
extern void malloc_shutdown(void);

.
diff -re crt/src/dllcrt0.c crt-sp1/src/dllcrt0.c
189,190d
136d
126d
111d
106,108d
diff -re crt/src/intel/_sample_.def crt-sp1/src/intel/_sample_.def
1198d
718,719d
664a
    posix_memalign
.
638a
    memalign
.
495a
    malloc_usable_size
.
331,336d
319d
313,314d
81d
9c
LIBRARY MOZCRT19
.
diff -re crt/src/internal.h crt-sp1/src/internal.h
413a
#endif
.
409c
#if 0
.
diff -re crt/src/makefile crt-sp1/src/makefile
1771a
$(DEFFILE_DIR)\$(RETAIL_LIB_NAME).def : $(DEFFILE_DIR)\_sample_.def
	copy $** $@
.
1235,1237d
760c
dll_ :: $(OBJROOT) $(OBJCPUDIR) $(OBJDIR_DLL) $(OBJDIR_DLL)\$(PURE_OBJ_DIR) \
   	$(OBJDIR_DLL)\$(CPP_OBJ_DIR) \
   	$(RELDIR_CPU) $(PDBDIR_CPU_DLL) $(MAKE_DIRS_DLL)
.
340c
CC_OPTS_BASE=-c -nologo -Zlp8 -W3 -GFy -DWIND32
.
213d
21,22c
RETAIL_DLL_NAME=MOZCRT19
RETAIL_LIB_NAME=msvcrt
.
diff -re crt/src/makefile.inc crt-sp1/src/makefile.inc
1635a

$(OBJDIR)\unhandld.obj: $(PREOBJDIR)\unhandld.obj
        copy $(PREOBJDIR)\unhandld.obj $@
.
623d
621d
405d
344,356c
        $(OBJDIR)\jemalloc.obj  \
.
336,337d
331,332d
329d
325d
321,322d
diff -re crt/src/makefile.sub crt-sp1/src/makefile.sub
66c
CFLAGS=$(CFLAGS) -O2 -DMOZ_MEMORY=1 -DMOZ_MEMORY_WINDOWS=1
.
diff -re crt/src/malloc.h crt-sp1/src/malloc.h
189a
#endif
.
177a

#if 0
.
83a
#endif
.
70a
#if 0
.
diff -re crt/src/mlock.c crt-sp1/src/mlock.c
274c
#endif
.
262a
#if 0
.
diff -re crt/src/new.cpp crt-sp1/src/new.cpp
60d
52,55d
37,38c
        break;
.
diff -re crt/src/nothrownew.cpp crt-sp1/src/nothrownew.cpp
38c
#endif
.
31a
#if 1
        break;
#else

.
diff -re crt/src/tidtable.c crt-sp1/src/tidtable.c
360,361d
